src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -27,123 +26,38 @@ namespace Microsoft.Identity.Client.Http
-        {
-            return SendRequestAsync(endpoint, headers, null, HttpMethod.Get, logger, doNotThrow: true, cancellationToken: cancellationToken);
-        }
-
-        /// <summary>
-        /// Performs the POST request just like <see cref="SendPostAsync(Uri, IDictionary{string, string}, IDictionary{String, String}, ILoggerAdapter, CancellationToken)"/>
-        /// but does not throw a ServiceUnavailable service exception. Instead, it returns the <see cref="HttpResponse"/> associated
-        /// with the request.
-        /// </summary>
-        public virtual Task<HttpResponse> SendPostForceResponseAsync(
-            Uri uri,
-            IDictionary<string, string> headers,
-            IDictionary<string, string> bodyParameters,
-            ILoggerAdapter logger,
-            CancellationToken cancellationToken = default)
+        public HttpManager(
+            IMsalHttpClientFactory httpClientFactory,
+            Func<HttpResponse, bool> retryCondition)
         {
-            HttpContent body = bodyParameters == null ? null : new FormUrlEncodedContent(bodyParameters);
-            return SendRequestAsync(uri, headers, body, HttpMethod.Post, logger, doNotThrow: true, cancellationToken: cancellationToken);
-        }
-
-        /// <summary>
-        /// Performs the POST request just like <see cref="SendPostAsync(Uri, IDictionary{string, string}, HttpContent, ILoggerAdapter, CancellationToken)"/>
-        /// but does not throw a ServiceUnavailable service exception. Instead, it returns the <see cref="HttpResponse"/> associated
-        /// with the request.
-        /// </summary>
-        public virtual Task<HttpResponse> SendPostForceResponseAsync(
-            Uri uri,
-            IDictionary<string, string> headers,
-            StringContent body,
-            ILoggerAdapter logger,
-            CancellationToken cancellationToken = default)
-        {
-            return SendRequestAsync(uri, headers, body, HttpMethod.Post, logger, doNotThrow: true, cancellationToken: cancellationToken);
-        }
-
-        private static HttpRequestMessage CreateRequestMessage(Uri endpoint, IDictionary<string, string> headers)
-        {
-            HttpRequestMessage requestMessage = new HttpRequestMessage { RequestUri = endpoint };
-            requestMessage.Headers.Accept.Clear();
-            if (headers != null)
-            {
-                foreach (KeyValuePair<string, string> kvp in headers)
-                {
-                    requestMessage.Headers.Add(kvp.Key, kvp.Value);
-                }
-            }
-
-            return requestMessage;
+            _httpClientFactory = httpClientFactory ??
+                throw new ArgumentNullException(nameof(httpClientFactory));
+            _retryCondition = retryCondition;
         }
 
-        protected virtual async Task<HttpResponse> SendRequestAsync(
+        public async Task<HttpResponse> SendRequestAsync(
```

Comment:
One thin to verify here is that this behavior matches the "NoRetry" behavior of the previous HttpManager class, especially exception handling. For ex, what if the request times out? Previously we'd throw RequestTimeout MsalServiceException; looks like the new behavior is the same; but since there were a few code changes, I suggest to double check all scenarios.

https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/d9ac2c1483befc1208b7244fda086005391ccde3/src/client/Microsoft.Identity.Client/Http/HttpManager.cs#L180